src/csl/rendering/mod.rs

```diff
@@ -312,25 +335,36 @@ impl RenderCsl for citationberg::Number {
         ctx.commit_elem(
             depth,
             self.display,
-            (self.variable == NumberVariable::CitationNumber)
-                .then_some(ElemMeta::CitationNumber)
-                .or(Some(ElemMeta::Number)),
+            Some(
+                if self.variable
+                    == NumberOrPageVariable::Number(NumberVariable::CitationNumber)
+                {
+                    ElemMeta::CitationNumber
+                } else {
+                    ElemMeta::Number
+                },
+            ),
         );
     }
 
     fn will_render<T: EntryLike>(&self, ctx: &mut Context<T>, var: Variable) -> bool {
         match ctx.instance.kind {
-            Some(SpecialForm::VarOnly(Variable::Number(n))) if self.variable != n => {
+            Some(SpecialForm::VarOnly(Variable::Number(n)))
+                if self.variable != NumberOrPageVariable::Number(n) =>
+            {
                 return false
             }
             Some(SpecialForm::OnlyFirstDate | SpecialForm::OnlyYearSuffix) => {
-                return self.variable == NumberVariable::Locator
+                return matches!(
+                    self.variable,
+                    NumberOrPageVariable::Number(NumberVariable::Locator)
+                )
             }
             Some(SpecialForm::VarOnly(_)) => return false,
             _ => {}
         }
```

Comment:
I feel like something is off in this function. Seems like VarOnly forms will always return `false` here, even if the variables match.

My change at L353 is actually wrong since, in theory, a matching page variable should also pass. But since VarOnly always fails, it doesn't matter. However, since it appears that VarOnly is currently only used for author variables, this potential bug has no practical effect. Either way, do we know if this is intentional? @reknih

Could always fix it later, but just curious as i just noticed this. :)

Reply:
Since this effectively isn't a regression, I'll be merging this and we can fix in a separate PR if there is a bug here.